### PR TITLE
Loki: Add optional stream selector to fetchLabelValues API

### PR DESCRIPTION
**Pull request by ivanahuckova**
In this PR, we are adding optional options to `fetchLabelValues` with possibility to specify `streamSelector`. The API for fetching label values includes options to add `query` which is basically `streamSelector` (I decided to name parameter `streamSelector` cause `query` is vague and we can't include things like parsers or other operations) https://grafana.com/docs/loki/latest/reference/api/#list-label-values-within-a-range-of-time

The reason why we are creating `options` rather than just passing `streamSelector` as second parameter is because in the future we could expand this to other things, such as specifying custom `timeRange` or something else. 

This was requested by @trevorwhitney to help them develop app plugin.

### CODE REVIEW

File: public/app/plugins/datasource/loki/docs/app_plugin_developer_documentation.md [hunk 1]
```diff
@@ -58,13 +58,16 @@ The `datasource.languageProvider.fetchLabelValues()` method is designed for fetc
  * It returns a promise that resolves to an array of strings containing the label values.
  *
  * @param labelName - The name of the label for which you want to retrieve values.
+ * @param options - (Optional) An object containing additional options - currently only stream selector.
+ * @param options.streamSelector - (Optional) The stream selector to filter label values. If not provided, all label values are fetched.
+ * @returns A promise containing an array of label values.
```
Comment (svennergr) [inline public/app/plugins/datasource/loki/docs/app_plugin_developer_documentation.md:63]:
```suggestion
```

**Review by svennergr — APPROVED** [review]
Just a nit, rest LGTM!

File: public/app/plugins/datasource/loki/LanguageProvider.test.ts [hunk 2]
```diff
@@ -237,6 +237,15 @@ describe('Language completion provider', () => {
       expect(labelValues).toEqual(['label1_val1', 'label1_val2']);
     });
 
+    it('fetch label when options.streamSelector provided and values is not cached', async () => {
```
Comment (trevorwhitney) [inline public/app/plugins/datasource/loki/LanguageProvider.test.ts:240]: what's this test testing? I notice it still passes if I comment all the new `streamSelector` and `streamParam` logic in `fetchLabelValues`

File: public/app/plugins/datasource/loki/LanguageProvider.test.ts [hunk 3]
```diff
@@ -250,6 +259,19 @@ describe('Language completion provider', () => {
       expect(nextLabelValues).toEqual(['label1_val1', 'label1_val2']);
     });
 
+    it('should return cached values when options.streamSelector provided', async () => {
```
Comment (trevorwhitney) [inline public/app/plugins/datasource/loki/LanguageProvider.test.ts:262]: same with this test, passes with all `streamParam` logic commented out

**Review by trevorwhitney — COMMENTED** [review]
Looks good, though I noticed a few of the tests don't seem to be testing the new logic. We're either missing the correct assertions or they're not necessary.

File: public/app/plugins/datasource/loki/LanguageProvider.test.ts (hunk 2 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/LanguageProvider.test.ts:243]: It was testing that the request is successfully called and we receive correct values, but you are right, we weren't testing if `streamSelector` was actually included in request. I've updated both tests to correctly check if streamSelector was added. Thanks for catching this!

**Pull request merged**